src/components/Workspace.jsx

```diff
@@ -128,6 +131,24 @@ class Workspace extends React.Component {
     addEventListener('beforeunload', this._confirmUnload);
   }
 
+  componentDidUpdate(prevProps) {
```

Comment:
@outoftime I was thinking about this during my commute this morning and it occurred to me that it might be cleaner to manage the entire lifecycle of the gist export window (currently `_handleExportGist`, `_loadExportedGist`, and `_cleanUpGistExport`) within the `exportGist` saga (or something like "displayGist" saga). 

`Workspace` and the gist export window have no inherent relationship (especially not with the refactoring you've already got here), and distancing them will 
1. make that clearer
2. remove the need for this lifecycle hook (comparing with prev props is a minor code smell IMHO).